cmd/root.go

```diff
@@ -3,242 +3,237 @@ package cmd
 import (
 	"fmt"
 
-	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
-	"github.com/spf13/viper"
 
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/config"
 	"github.com/spacemeshos/go-spacemesh/config/presets"
 	"github.com/spacemeshos/go-spacemesh/node/flags"
 )
 
-var cfg = config.MainnetConfig()
+func AddFlags(flagSet *pflag.FlagSet, cfg *config.Config) (configPath *string) {
+	// A workaround to keep the original config intact to avoid
+	// overwriting it with the default values.
+	original := *cfg
+	defer func() { *cfg = original }()
```

Comment:
I don't understand; doesn't this completely undo every flag that was just parsed?